syntax = "proto3";

package events;

service EventsService {
    rpc CreateEvent(CreateEventRequest) returns (EventResponse);
    rpc GetEvent(GetEventRequest) returns (EventResponse);
    rpc ListEvents(ListEventsRequest) returns (EventsListResponse);
    rpc RegisterForEvent(RegisterRequest) returns (RegisterResponse);
    rpc GetRegistrations(GetRegistrationsRequest) returns (RegistrationsResponse);
    rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

message Event {
    int32 id = 1;
    string title = 2;
    string description = 3;
    string type = 4;  // conference, lecture, workshop, etc.
    string location = 5;
    string start_time = 6;
    string end_time = 7;
    int32 max_participants = 8;
    int32 registered_count = 9;
    int32 created_by = 10;
    string created_at = 11;
}

message Registration {
    int32 id = 1;
    int32 event_id = 2;
    int32 user_id = 3;
    string user_name = 4;
    string registered_at = 5;
}

message CreateEventRequest {
    string title = 1;
    string description = 2;
    string type = 3;
    string location = 4;
    string start_time = 5;
    string end_time = 6;
    int32 max_participants = 7;
    int32 created_by = 8;
}

message GetEventRequest {
    int32 id = 1;
}

message ListEventsRequest {
    int32 page = 1;
    int32 page_size = 2;
    string type = 3;
    string start_date = 4;
    string end_date = 5;
}

message RegisterRequest {
    int32 event_id = 1;
    int32 user_id = 2;
}

message GetRegistrationsRequest {
    int32 event_id = 1;
}

message EventResponse {
    Event event = 1;
    string message = 2;
}

message EventsListResponse {
    repeated Event events = 1;
    int32 total_count = 2;
}

message RegisterResponse {
    bool success = 1;
    string message = 2;
}

message RegistrationsResponse {
    repeated Registration registrations = 1;
    int32 total_count = 2;
}

message HealthCheckRequest {}
message HealthCheckResponse {
    string status = 1;
    string version = 2;
    string timestamp = 3;
}

